lib/zendesk_apps_support/validations/mime.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module ZendeskAppsSupport
+  module Validations
+    module Mime
+      UNSUPPORTED_MIME_TYPES = %w[
+        vnd.rar rar zip gzip pdf doc docx avi bin bz bz2 csh sh jar mp3 mpeg odt pptx ppt xls xlsx 7z
+      ].freeze
+
+      class << self
+        def call(package)
+          unsupported_files = package.files.find_all do |app_file|
+            identify_mime_type(app_file) && block_listed?(app_file)
+          end.map(&:relative_path)
+
+          [mime_type_warning(unsupported_files)] if unsupported_files.any?
+        end
+
+        private
+
+        def block_listed?(app_file)
+          app_file_mime_type = identify_mime_type(app_file)
+
+          file_extension = app_file_mime_type.extension
+          content_subtype = app_file_mime_type.content_type.split('/').last
+
+          ([file_extension, content_subtype] & UNSUPPORTED_MIME_TYPES).any?
+        end
+
+        def identify_mime_type(app_file)
+          require 'mini_mime'
```

Comment:
nit: I prefer to have this at the top of the file, so you don't have to read through the code to see dependencies

Reply:
In fact, could this method just be collapsed into `block_listed?`, since it's just an alias for a `MiniMime` method?

Reply:
YES we can! It doesn't serve much purpose being extracted out as a private method anyway. Love the suggestion!!!